src/stats.c

```diff
@@ -42,6 +42,15 @@ const char *arena_mutex_names[mutex_prof_num_arena_mutexes] = {
 	assert(miblen_new == miblen + 1);				\
 } while (0)
 
+#define CTL_M1_GET(n, i, v, t) do {					\
+	size_t mib[CTL_MAX_DEPTH];					\
+	size_t miblen = sizeof(mib) / sizeof(size_t);			\
+	size_t sz = sizeof(t);						\
+	xmallctlnametomib(n, mib, &miblen);				\
+	mib[1] = (i);							\
+	xmallctlbymib(mib, miblen, (void *)v, &sz, NULL, 0);		\
+} while (0)
+
```

Comment:
Try merging  CTL_M1_GET and _M2_GET. You can define a CTL_MIB_GET which takes the index and shared by the M1 and M2 getters.

Reply:
Merged.